routes/conversation.js

```diff
@@ -255,6 +255,39 @@ exports.leave = function (req, res){
 	});
 };
 
+/**
+ * Action when a user archives a conversation.
+ */
+exports.archive = function (req, res){
+	if (!req.user || !req.user._id){
+		resError(res, "Access denied.", "/error");
+		return;
+	}
+
+	// Search through the user's conversation list until the conversation with the correct
+	// id is found. 
+	User.findById( req.user._id, function (err, user){
+		if (err){
+			reject(res, "Could not find the user.");
+			return;
+		}
+
+		// Save the id of the conversation that we want to archive.
+		var targetConversationId = JSON.stringify(req.body.conversationId);
+
+		// Find the index at which the target conversation exists amongst all of the user's conversations.
+		var conversationIds = user.userConversations.map(function(p) {return p.conversation});
+		var conversationIdStrings = conversationIds.map(function(id) {return JSON.stringify(id)});
+		var found = conversationIdStrings.indexOf(targetConversationId) !== -1;
+
+		// Modify the hallOfFame boolean associated with the conversation to true if it exists.
+		if (found) {
+			// Something along the lines of: user.userConversations[found].hallOfFame == true;
```

Comment:
I think your pseudo-code is the right way to start.  Although, it seems that the `found` variable is boolean, but if you have the `foundIndex`, you can indeed do `user.userConversations[foundIndex].hallOfFame == true;`, similar to as you suggested.

Once the value of the object is set to true, all that's left is to save the modified object in the database.  You could find an example of something like that in "routes/user.js" around line 38.  So, in the archive function, once we find the conversation in the user's list, you could do something like
```
user.save(function(err){
		if (err){
			return resError(res, "...some error message...");
		}

		/* Insert code that modified the isThrilled inside the convo object */
	});
```

Then, inside the save callback, if there is no error, we still need to update the convo object, which has a list of participants including whether or not each one is thrilled.  So in that place, you would need to: find the conversation using the conversationId you got before, update the participants property, and save the conversation.  The code flow could look like:
```
Conversation.findById(convoId, function(err, convo){
	if (err) return resError(res, "could not find convo")

	/* Do some regular JS that updates the convo.participants array to set the isThrilled to true on the correct user */

	// Save the modified conversation.
	convo.save(function(err){
		if (err) return resError(res, "could not save the convo")

		// Finally everything is done, so send the response back the the user with success.
		res.send({success: true, redirect: '/home'});
	});
});
```

Let me know if you have any questions at all!  I realize the code I wrote for some other functions isn't the must readable, so I can explain other functions if you need as well.  Thanks for taking this on, Nikil!